### ISSUE TITLE
Already has 1 active runs, not queuing any tasks for run

### ISSUE THREAD

**Issue by akotlar** [apache/airflow#13198]
**Apache Airflow version**: 2.0.0

**Environment**:

- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release): Ubuntu 18.04
- **Kernel** (e.g. `uname -a`): Linux workstation38 5.4.0-52-generic #57~18.04.1-Ubuntu SMP Thu Oct 15 14:04:49 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux

**What happened**:

Ran a dag, could not trigger a subsequent run.

Dag settings:

```
with DAG(dag_id=DAG_NAME,
         default_args=default_args,
         schedule_interval='@daily',
         max_active_runs=1,
         catchup=False,
         tags=['tag'],
         is_paused_upon_creation=True) as dag:
```

Dag ran, and was marked as successful. Triggering a 2nd run (manually) failed to schedule any tasks, with the message in scheduler logs:

[2020-12-20 16:05:50,919] {scheduler_job.py:1667} INFO - DAG <dag_name> already has 1 active runs, not queuing any tasks for run 2020-12-20 20:52:38.216778+00:00

To complicate things: during the execution of the dag, I modified the name of one of the tasks, and this was done, I believe, after that task had finished processing. This resulted in further UI / log location issues described below.

**What you expected to happen**:

The manually triggered DAG should have resulted in tasks being scheduled.

The DAG was somehow not entirely marked as successful, hitting max_active_runs limits. 

The fact that I had changed the name of a task inside of a task group, before the dag finished, may have screwed up the check.  Interestingly, the UI showed the first dag run as completed.

**How to reproduce it**:

```
with DAG(dag_id=DAG_NAME,
         default_args=default_args,
         schedule_interval='@daily',
         max_active_runs=1,
         catchup=False,
         tags=['tag'],
         is_paused_upon_creation=True) as dag:
    t1= task(function_one)
    t2= task(function_two)
    t3= task(function_three) # say I change the function name to function_three_2, the clobbering issue will be triggered
    t4= task(function_four)

    start = DummyOperator(task_id="start")
    end = DummyOperator(task_id="end")
    for abs_pivat_run_path in stable_folder_list:
        basename = os.path.basename(abs_pivat_run_path)
   
        with TaskGroup(basename, tooltip=f"Tasks for {basename}") as task_group:
            t1(path) >> t2() >> t3(abs_pivat_run_path) >> t4(abs_pivat_run_path)

        start >> task_group >> end

```

There also appear to be a UI issue. The tasks whose name changed had the new name reflected in the UI, but with what appeared to be the state of the original task. However, the log location was missing (pointed to the new log name, which hadn't re-run). I think this happens because somehow the task id of a task instance is being clobbered.

**Anything else we need to know**:

How often does this problem occur? Once? Every time etc?

Every time

**Comment by RikHeijdens** [comment]
Hi @akotlar, I believe this is the same issue that I've reported in https://github.com/apache/airflow/issues/13151. It ought to be fixed by https://github.com/apache/airflow/pull/13165.

**Comment by akotlar** [comment]
It looks that way! Thanks

**Comment by akotlar** [comment]
Closing, will reopen if #13165 doesn't fix.

**Comment by peschue** [comment]
I still have this problem and I run version 2.0.1. Maybe it is because of my special DAG setup: I have a DAG that runs each day, and an ExternalTaskSensor that waits for the previous day's completion and an External TaskSensor that waits for the previous week's completion (part of the DAG is skipped and runs only once a week).

If I clear many of the DAGs with the intention to let it catch up in the correct order (by waiting until one week is done and then doing the next week) I get flooded with "removed" task states from everything depending on the task sensors and the log says  "Already has 18 active runs, not queuing any tasks for run" (although max_concurrency is 100) and clearing the removed tasks just makes them re-appear. If I deactivate the DAG, I can clear the "removed" tasks and they appear only if I reactivate the DAG. Still, nothing gets scheduled so the task sensors have no chance to ever succeed.

**Comment by abhishekshenoy** [comment]
@peschue Were you able to get a resolution for this issue?

**Comment by peschue** [comment]
@abhishekshenoy I deactivated the dag and used the job search to find all 'removed' tasks and cleared them, then reactivated the dag. Recently I upgraded to 2.1.0-rc2 because of another issue. We didn't have this issue again so far.

**Comment by WesleyBatista** [comment]
I was having the same problem, but our dag has only one task.
Airflow version: 2.0.2

Following the insight from @peschue, from the Airflow UI:
- I just looked the past executions on the Tree View trying to find some task with a state different then _failure_ or _success_.
- I found one DagRun in the failed state (manually marked), but the task was without a state (no color); and another DagRun with a task in the _shutdown_ state (blue color)
- I marked the two tasks as failed.

The new DagRun started scheduling tasks.

Thanks community :pray:
